include/slang/driver/SourceLoader.h

```diff
@@ -159,6 +163,10 @@ class SLANG_EXPORT SourceLoader {
         // The filesystem path (as specified by the user).
         std::filesystem::path path;
 
+        // An optional pre-loaded buffer for when the source doesn't originate
+        // from the filesystem
+        const SourceBuffer preloadedBuffer;
```

Comment:
This shouldn't be marked const; typically const fields in data structures can make them difficult to work with, like no longer being movable for example, which can make containers of them less efficient.